### ISSUE TITLE
Frontlight auto-off

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#8997]
**Context:**
Due to their reflective nature, e-ink screens stay readable with very low levels of frontlight (I have never exceeded 15 percent on my Kobo Libra). From the threads I read, I got the impression that many people use theirs at lower levels too.

**Problem:**
After reading at night, while reading in the morning my frontlight generally stays on until I notice it from the status bar indicator (which I keep there only for this purpose). Because at such low frontlight levels, on/off state is indistinguishable under daylight so I don't notice when I start reading.

**Solution:**
Frontlight can switch off automatically at a user determined time
OR
This feature can be integrated with Auto Warmth plugin so it can switch off frontlight automatically at sunrise

Yeah I know, it's not a big problem but computers are better than us at automated tasks like this so if there are other people who have the same problem maybe this feature can be added.

**Comment by zwim** [comment]
What are the thoughts here to implement a frontlight ramp (idle time, duration)  in the powersaving (aka autosuspend) plugin?
That would have been on my agenda as well.

**Comment by zwim** [comment]
@offset-torque autowarmth might not be the right place, as rhia would be an power saving feature.

**Comment by Frenzie** [comment]
I wouldn't worry too much about whether it's currently Auto**Warmth** unless it's architecturally a really bad fit. Conceptually speaking all that stuff you wrote for sunset seems like it should fit in there?

**Comment by zwim** [comment]
I have a preliminary implementation of the automatic frontlight dimming ready. 

But I have one minor problem/question: I don't want that a tap is executed by the reader, when dimming is running. Instead of that, the first tap should just stop the dimming. It does not help to set the return value of `onInputEvent` to true.
I can modify `UIManager` and `hook_container` to stop evaluation events, when the first event delivers true. But I am wondering if there isn't a more elegant solution?

In short: I want that the behavior describedin http://koreader.rocks/doc/topics/Events.md.html also holds true for `onInputEvent`.

**Comment by NiLuJe** [comment]
Wouldn't that be a job for the discardEvent thingy?

**Comment by JJTech0130** [comment]
Is this still not implemented? Seems like a useful feature to have.

**Comment by zwim** [comment]
It is int he making see #9337.

**Comment by offset-torque** [comment]
Feature implemented. Closing the issue. Thanks @zwim.
